File: scala-finch/src/main/scala/org/openapitools/models/Presence.scala
package org.openapitools.models

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import org.openapitools._

/**
 * `{client_name}`: Object containing the details of the user's presence on a particular platform with the client's platform name being the object key. 
 * @param client The client's platform name. 
 * @param status The status of the user on this client. It is either `idle` or `active`. 
 * @param timestamp The UNIX timestamp of when this client sent the user's presence to the server with the precision of a second. 
 * @param pushable Whether the client is capable of showing mobile/push notifications to the user. 
 */
case class Presence(client: Option[String],
                status: Option[String],
                timestamp: Option[Int],
                pushable: Option[Boolean]
                )

object Presence {
    /**
     * Creates the codec for converting Presence from and to JSON.
     */
    implicit val decoder: Decoder[Presence] = deriveDecoder
    implicit val encoder: ObjectEncoder[Presence] = deriveEncoder
}

File: php-laravel/lib/app/Models/Presence.php
<?php
/**
 * Presence
 */
namespace app\Models;

/**
 * Presence
 */
class Presence {

    /** @var string $client The client&#39;s platform name.*/
    private $client;

    /** @var string $status The status of the user on this client. It is either &#x60;idle&#x60; or &#x60;active&#x60;.*/
    private $status;

    /** @var int $timestamp The UNIX timestamp of when this client sent the user&#39;s presence to the server with the precision of a second.*/
    private $timestamp;

    /** @var bool $pushable Whether the client is capable of showing mobile/push notifications to the user.*/
    private $pushable;

}
